src/core/database/database.service.ts

```diff
@@ -186,6 +241,20 @@ export class DatabaseService {
     changes: { [Key in keyof TObject]?: UnwrapSecured<TObject[Key]> };
   }) {
     let updated = object;
+    for (const prop of props) {
+      if (
+        changes[prop] === undefined ||
+        unwrapSecured(object[prop]) === changes[prop]
+      ) {
+        continue;
+      }
+      if (!(await this.authorizationService.checkIfCanEdit(object, prop))) {
+        throw new UnauthorizedException(
+          'You do not have permission to update this property',
+          prop
+        );
+      }
+    }
```

Comment:
I thought we said we wanted to keep authorization checks out of this class.

This could be an easy
```ts
authorization.verifyCanEdit(currentObject, changes);
```
Which would conditionally throw an exception.